app/services/trainees/set_academic_cycles.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Trainees
+  class SetAcademicCycles
+    include ServicePattern
+
+    DURATIONS = %w[hours months days years].freeze
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      trainee.start_academic_cycle = start_academic_cycle
+      trainee.end_academic_cycle = end_academic_cycle
+
+      trainee.save!
```

Comment:
I wondered when we did the cohort one whether we really want to save the trainee inside the service or whether we should just set the start and end cycles and let the calling code save when it's ready. Still wondering.

Reply:
I also feel like updating the object in a service is not great - this code is called from within a job so I'm not sure it makes a huge difference if I do:

```
module Trainees
  class SetAcademicCyclesJob < ApplicationJob
    def perform(trainee)
      trainee = Trainees::SetAcademicCycles.call(trainee: trainee)
      trainee.save!
    end
  end
end
```

It definitely means the service _is_ only setting the cycles like it says it is! What do you think?

Reply:
Yeah. I think that's probably better. Then you're not inadvertently saving it before you're ready or saving it more than once if you need to do some other things (which we admittedly aren't doing but it seems like 'the right thing to do ™️'. 👍

Reply:
done!